### PR TITLE
added snowflake support and url obfuscation

### PR DESCRIPTION
**Pull request by mnbogner**
- update dependencies
 - add snowflake url and cleanup unused urls
 - revise analytics to include direct urls as selected url events
 - urls included in events and logging are now obfuscated
 - removed redundant CronetInterceptor

### CODE REVIEW

File: app/src/main/java/org/wikipedia/main/MainActivity.kt [hunk 1]
```diff
@@ -83,12 +81,16 @@ class MainActivity : SingleFragmentActivity<MainFragment>(), MainFragment.Callba
         override fun onReceive(context: Context?, intent: Intent?) {
             if (intent != null && context != null) {
                 if (intent.action == ENVOY_BROADCAST_VALIDATION_SUCCEEDED) {
-                    val validUrl = intent.getStringExtra(ENVOY_DATA_URL_SUCCEEDED)
-                    val validService = intent.getStringExtra(ENVOY_DATA_SERVICE_SUCCEEDED)
+                    val validUrl = intent.getStringExtra(ENVOY_DATA_URL_SUCCEEDED) ?: ""
+                    var validService = intent.getStringExtra(ENVOY_DATA_SERVICE_SUCCEEDED) ?: ""
+                    if (validUrl.startsWith("envoy") && validService.startsWith("http")) {
```
Comment (vitriolix) [inline app/src/main/java/org/wikipedia/main/MainActivity.kt]: why is the service http if the url is envoy?

File: app/src/main/java/org/wikipedia/main/MainActivity.kt [hunk 2]
```diff
@@ -83,12 +81,16 @@ class MainActivity : SingleFragmentActivity<MainFragment>(), MainFragment.Callba
...
+                        validService = "envoy"
+                    }
+                    val sanitizedUrl = UrlUtil.sanitizeUrl(validUrl, validService)
```
Comment (vitriolix) [inline app/src/main/java/org/wikipedia/main/MainActivity.kt:88]: is sanitizing related to snowflake changes somehow?

**Review by vitriolix — COMMENTED** [review]
the code looks fine

**Review by stevenmcdonald — APPROVED** [review]

**Comment by vitriolix** [comment]
🚢

File: app/src/main/java/org/wikipedia/main/MainActivity.kt (hunk 1 above)
Comment (mnbogner) [reply app/src/main/java/org/wikipedia/main/MainActivity.kt]: just to include this here: they're handled the same way in envoy so they get lumped together, but now i need to make the distinction for the revised analytics

File: app/src/main/java/org/wikipedia/main/MainActivity.kt (hunk 2 above)
Comment (mnbogner) [reply app/src/main/java/org/wikipedia/main/MainActivity.kt:88]: i may have overemphasized the snowflake aspect.  this pr is more about both updating the secrets (which includes adding snowflake) and obfuscating them better before the next release

**Comment by mnbogner** [comment]
revised this to use the new envoy version of the UrlUtil class: https://github.com/greatfire/envoy/pull/67  will need to make one more commit to update the dependencies when that version of envoy is published.

**Pull request merged**